### DISCUSSION TITLE
How to add test cases for Metrices

### DISCUSSION THREAD

**Discussion by RSoodAMAT** [open-telemetry/opentelemetry-java#3163] [Q&A]
I am using metrices in my system We need to write test cases for same to increase the code coverage . 
How do i write the same.

**Comment by jkwatson** [comment]
There are a couple of options that I see for testing metrics.

1) mock your Meter/Instruments and verify that the expected calls are being made.
2) Use the `io.opentelemetry.sdk.metrics.testing.InMemoryMetricsExporter` in the SDK that you use for testing and use it to verify that the output is what you want.

Hope that helps!
